RecoCTPPS/TotemRPLocal/interface/TotemRPLocalTrackFitterAlgorithm.h

```diff
@@ -0,0 +1,73 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors:
+* 	Hubert Niewiadomski
+*   Jan Kašpar (jan.kaspar@gmail.com)
+*
+****************************************************************************/
+
+#ifndef RecoCTPPS_TotemRPLocal_TotemRPLocalTrackFitterAlgorithm
+#define RecoCTPPS_TotemRPLocal_TotemRPLocalTrackFitterAlgorithm
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPRecHit.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPLocalTrack.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/TotemRPGeometry.h"
+#include "Geometry/VeryForwardRPTopology/interface/RPTopology.h"
+
+#include "TVector3.h"
+#include "TVector2.h"
+
+#include <unordered_map>
+
+//----------------------------------------------------------------------------------------------------
+
+/**
+ *\brief Algorithm for fitting tracks through a single RP.
+ **/
+class TotemRPLocalTrackFitterAlgorithm
+{
+  public:
+    TotemRPLocalTrackFitterAlgorithm(const edm::ParameterSet &conf);
+
+    /// performs the track fit, returns true if successful
+    bool FitTrack(const edm::DetSetVector<TotemRPRecHit> &hits, double z_0, const TotemRPGeometry &tot_geom, TotemRPLocalTrack &fitted_track);
+
+    /// Resets the reconstruction-data cache.
+    void Reset();
```

Comment:
The "reconstruction algebra objects" are build on request, thus in "FitTrack". It could be moved to "Reset" or "Init", but then it would need to build objects for all RPs. In TOTEM we typically do not use all RPs in one run, e.g. with the 2016 standard runs we use 4 out of 24 RPs. In terms of performance we are discussing peanuts, typically the Reset/Init is executed once a fill. 

Unless you have strong objections, I'd keep the code as is.